signalfx/resource_signalfx_aws_integration.go

```diff
@@ -509,30 +528,16 @@ func integrationAWSCreate(d *schema.ResourceData, meta interface{}) error {
 	if err := d.Set("name", preInt.Name); err != nil {
 		return err
 	}
-	payload, err := getPayloadAWSIntegration(d)
-	if err != nil {
-		return fmt.Errorf("Failed creating json payload: %s", err.Error())
-	}
-
-	debugOutput, _ := json.Marshal(payload)
-	log.Printf("[DEBUG] SignalFx: Create (Update) AWS Integration Payload: %s", string(debugOutput))
-
-	int, err := config.Client.UpdateAWSCloudWatchIntegration(context.TODO(), d.Get("integration_id").(string), payload)
-	if err != nil {
-		if strings.Contains(err.Error(), "40") {
-			err = fmt.Errorf("%s\nPlease verify you are using an admin token when working with integrations", err.Error())
-		}
-		return err
-	}
-	d.SetId(int.Id)
+	d.SetId(preInt.Id)
 
-	return awsIntegrationAPIToTF(d, int)
+	return integrationAWSUpdate(d, meta)
 }
 
 func integrationAWSUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*signalfxConfig)
 
-	payload, err := getPayloadAWSIntegration(d)
+	useMetricStreamsSyncHasChange := d.HasChange("use_metric_streams_sync")
+	payload, err := getPayloadAWSIntegration(d, useMetricStreamsSyncHasChange)
```

Comment:
I'd rather keep `getPayloadAWSIntegration` args list short as we can read the `d.HasChange(...)` inside the method.